src/core/server/http/lifecycle_handlers.test.ts

```diff
@@ -0,0 +1,275 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+  createCustomHeadersPreResponseHandler,
+  createVersionCheckPostAuthHandler,
+  createXsrfPostAuthHandler,
+} from './lifecycle_handlers';
+import { httpServerMock } from './http_server.mocks';
+import { HttpConfig } from './http_config';
+import { KibanaRequest, RouteMethod } from './router';
+
+const createConfig = (partial: Partial<HttpConfig>): HttpConfig => partial as HttpConfig;
+
+const forgeRequest = ({
```

Comment:
is there a reason not to use already existing fixtures ?https://github.com/elastic/kibana/blob/a84e4f5ab33498b2fe909b389214388d05eb1f23/src/core/server/http/http_server.mocks.ts#L97

Reply:
No, just forget we already had a mock request fixture. Will adapt.